datafusion/src/physical_plan/hash_join.rs

```diff
@@ -751,12 +788,19 @@ fn build_join_indexes(
 }
 
 macro_rules! equal_rows_elem {
-    ($array_type:ident, $l: ident, $r: ident, $left: ident, $right: ident) => {{
+    ($array_type:ident, $l: ident, $r: ident, $left: ident, $right: ident, $null_equal_null: ident) => {{
         let left_array = $l.as_any().downcast_ref::<$array_type>().unwrap();
         let right_array = $r.as_any().downcast_ref::<$array_type>().unwrap();
 
         match (left_array.is_null($left), right_array.is_null($right)) {
             (false, false) => left_array.value($left) == right_array.value($right),
+            (true, true) => {
+                if $null_equal_null {
```

Comment:
This can be `!$null_equal_null` instead (without if/else)

Reply:
Good point, I think you mean `$null_equal_null` not `!$null_equal_null` 😁

Reply:
Yes, I thought true/false were swapped in if/else, but probably I had that incorrect 👍